.github/workflows/application-onboarding-or-promotion-automation.yml

```diff
@@ -0,0 +1,95 @@
+name: Automation for application onboarding or promotion
+
+on:
+  workflow_dispatch:
+    inputs:
+      application_name:
+        description: Application name
+        required: true
+        type: string
+      application_environment:
+        description: Application environment
+        required: true
+        type: string
+      maintainer_github_user:
+        description: Maintainer GitHub username
+        required: true
+        type: string
+      task:
+        description: Onboarding or promoting
+        required: true
+        type: string
+  workflow_call:
+    inputs:
+      application_name:
+        description: Application name
+        required: true
+        type: string
+      application_environment:
+        description: Application environment
+        required: true
+        type: string
+      maintainer_github_user:
+        description: Maintainer GitHub username
+        required: true
+        type: string
+      task:
+        description: Onboarding or promoting
+        required: true
+        type: string
+
+jobs:
+  application-onboarding-or-promotion:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out ArgoCD infra repo
+        uses: actions/checkout@v2
+        with:
+          repository: department-of-veterans-affairs/vsp-infra-argocd
+          token: ${{ secrets.VA_VSP_BOT_GITHUB_TOKEN }}
+          path: vsp-infra-argocd
+
+      - name: Check if application already exists
+        id: exists
+        run: |
+          application_exists=$(cat vsp-infra-argocd/chart/values.yaml | yq -o=json | jq -e '.projects[] | select(.name == "${{ inputs.application_name }}").environments | any(. == "${{ inputs.application_environment }}")' || true)
+          if [ -z "$application_exists" ]; then export application_exists=false; fi
+          echo "::set-output name=application_exists::$(echo ${application_exists})"
+        
+      - name: If application exists
+        if: steps.exists.outputs.application_exists == 'true'
+        uses: KeisukeYamashita/create-comment@v1
```

Comment:
[peter-evans/create-or-update-comment](https://github.com/peter-evans/create-or-update-comment) has more stars in GitHub than the one you're using. We're using it to comment on the SOCKS access requests here: https://github.com/department-of-veterans-affairs/devops/blob/master/.github/workflows/pst-amt-roster-check.yml#L105 Should we try to go with a more widely-used action?

Reply:
This was just copy/pasted from the existing work Matt had already done. I have no strong opinions and not married to one action over the other.  I could go either way and don't care much as long as it does what we need it to do.

Reply:
oh, I didn't realize Matt was already using this action. I had come across this guidance in regards to gh Actions: https://vfs.atlassian.net/wiki/spaces/OT/pages/1474595236/Guidance+for+Github+Actions+usage so I always try to at least see how active the action is. https://github.com/KeisukeYamashita/create-comment hasn't been modified in two years and is not used by many projects. using `create-comment` feels like more of a vulnerability than `create-or-update-comment`, but i haven't read through the code of either...

Reply:
Fair. I will look into swapping it out.